src/Microsoft.Azure.SignalR.Management/Resilient/TimeoutHttpMessageHandler.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.SignalR.Management;
+
+#nullable enable
+
+internal class TimeoutHttpMessageHandler : DelegatingHandler
+{
+    private readonly TimeSpan _timeout;
+    public TimeoutHttpMessageHandler(IOptions<ServiceManagerOptions> serviceManagerOptions)
+    {
+        var options = serviceManagerOptions.Value;
+        if (options.RetryOptions == null)
+        {
+            // Timeout handled by HttpClient for backward compatibility
+            _timeout = Timeout.InfiniteTimeSpan;
```

Comment:
use another boolean to configure enable the handler or not?

Reply:
fixed